### DISCUSSION TITLE
Zero Page memory not used by DurexForth

### DISCUSSION THREAD

**Discussion by AussieSmitty** [jkotlinski/durexforth#354] [Q&A] [answered]
I see from the memory map in  the DurexForth v3.0.0 Operators Manual that many zero page locations are used for parameter stack and work areas for code words. If I want to temporarily use some zero page addresses for some assembler calls, what is safe to use? From what I can tell, it looks like $fb to $fe should be fine. Perhaps even $73 to $8a if one is not going to use any BASIC ROM functions (the same for $ff).

Can someone please advise as I'm keen to avoid having code that cashes, thanks.

**Comment by jkotlinski** [comment, marked as answer]
Hi Steve,
I would recommend the locations defined by W, W2 and W3.
Beyond that it’s on your own risk :-)

**Reply by AussieSmitty** [reply]
Hi jkotlinski,

Thanks. I think you're suggesting to avoid locations defined by W, W2 and W3, right? The manual says W is at $8b - $8c, W2 is at $8d - $8e, and W3 is at $9e - $9f. I'm guessing I should not write values into these memory locations.

So $fb to $ff should be safe to use without upsetting DurexForth?

Thanks,
Steve

**Reply by Whammo** [reply]
No Smitty,
  W, W2 and W3 Are guaranteed safe to use. They are free to use within your assembly routines.
 Still, do not store things there outside of each word, or between words.
  There may be other zero page locations that are free, but they are undocumented.

**Reply by AussieSmitty** [reply]
Hi Whammo,

I guess that so long as I only use these zero page areas within my assembly routines and not expect values to be there after using Forth words, then I'm safe.

Thanks

**Comment by Whammo** [comment]
I believe that's the long and short of it. It's good to see you're still at it. I'm glad to help.

**Reply by AussieSmitty** [reply]
Thanks Whammo,
I'm very new to this level of programming, so thanks so much for your assistance. With your help and my perserverance I've finally got RS232 starting to work in both directions under byte level control of Forth. This opens the door to having the C64 (under DurexForth) able to control numerous external devices and get feedback from these as well.
I'll update as I make more progress and have something interesting to show.
Again, I appreciate the assistance and it has helped me stick to it.
Smitty

**Reply by AussieSmitty** [reply]
All good Whammo. I'm using $fb to $fe for now and this may be all I need. From what I can tell these Zero Page locations are not used by DurexForth.

**Comment by Whammo** [comment]
What also seems to be free (unofficially) is CHRGET: $73 - $8A.
 
  The Forth word BYE implies that DurexForth is intended to exit to BASIC cleanly, and using CHRGET memory will disallow this, but it's pretty simple to reinitialize.

**Reply by AussieSmitty** [reply]
Thanks Whammo. This may well come in handy if I need more zero page use in the future.
